pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1109,7 +1116,35 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 		return err
 	}
 
+	addressesToDnat, err := b.getDstAddressesToDnat(proto)
+	if err != nil {
+		return err
+	}
+
 	if len(b.iface.Ports) == 0 {
+		if hasIstioSidecarInjectionEnabled(b.vmi) {
+			err = b.skipForwardingForPortsUsingNftables(proto, portsUsedByIstio())
+			if err != nil {
+				return err
+			}
+
+			err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND",
+				b.handler.GetNFTIPString(proto), "saddr", b.getSrcAddressesToSnat(proto),
+				"counter", "snat", "to", b.getGatewayByProtocol(proto))
+			if err != nil {
+				return err
+			}
+
+			err = b.handler.NftablesAppendRule(proto, "nat", "output",
+				b.handler.GetNFTIPString(proto), "daddr", addressesToDnat,
+				"counter", "dnat", "to", b.geVmIfaceIpByProtocol(proto))
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}
+
 		err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
```

Comment:
Shouldn't this rule be added only `if !hasIstioSidecarInjectionEnabled`?
Oh, you have `return nil` in the end of the previous if. It is confusing...

Reply:
Now it should be better.